lib/graphql/schema/implementation/method_call_implementation.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+module GraphQL
+  class Schema
+    class Implementation
+      # Turn the GraphQL field resolution into a Ruby method call
+      class MethodCallImplementation
+        def initialize(method:, graphql_arguments:, special_arguments:)
+          @method_name = method
+          @graphql_arguments = graphql_arguments
+          @special_arguments = special_arguments
+          @no_arguments = graphql_arguments.length == 0 && special_arguments.length == 0
+        end
+
+        def call(proxy, args, ctx)
+          if @no_arguments
+            proxy.public_send(@method_name)
+          else
+            method_args = {}
+            @graphql_arguments.each { |a| method_args[a] = args[a] }
+            @special_arguments.each do |arg|
+              case arg
+              when :context
+                method_args[:context] = ctx
+              when :irep_node
+                method_args[:irep_node] = ctx.irep_node
+              when :ast_node
+                method_args[:ast_node] = ctx.ast_node
+              end
+            end
+
+            proxy.public_send(@method_name, **method_args)
```

Comment:
This would probably be over-optimization at this point of development, but I wonder if eventually the `call` method here could be constructed at build-time by constructing the method as a string and then `instance_eval`'ing it. The idea is to avoid constructing the hash and doing this logic here - it may not even make a difference on performance, though.

Something like:
```ruby

call_args = @graphql_arguments.map { |a| "#{a}: args[:#{a}]" }.join(', ')
# "letter: args[:letter]"

method_body = <<~RUBY
  def call(proxy, args, ctx)
    proxy.#{@method_name}(#{call_args})
  end
RUBY

# def call(proxy, args, ctx)
#   proxy.suit(letter: args[:letter])
# end

instance_eval(method_body)
```